packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -130,9 +130,81 @@ const toggleBlock = (editor, value) => {
   Transforms.setNodes(editor, newProperties);
 };
 
-const BlocksDropdown = () => {
+const ALLOWED_MEDIA_TYPE = 'images';
+
+const imageSchemaFields = [
+  'name',
+  'alternativeText',
+  'url',
+  'caption',
+  'width',
+  'height',
+  'formats',
+  'hash',
+  'ext',
+  'mime',
+  'size',
+  'previewUrl',
+  'provider',
+  'provider_metadata',
+  'createdAt',
+  'updatedAt',
+];
+
+const pick = (object, keys) => {
+  return keys.reduce((acc, key) => {
+    if (object && Object.prototype.hasOwnProperty.call(object, key)) {
+      acc[key] = object[key];
+    }
+
+    return acc;
+  }, {});
+};
+
+const ImageDialog = ({ handleClose }) => {
+  const editor = useSlate();
+  const { components } = useLibrary();
+  const MediaLibraryDialog = components['media-library'];
+
+  const insertImages = (images) => {
+    images.forEach((img) => {
+      const image = { type: 'image', image: img, children: [{ type: 'text', text: '' }] };
+      Transforms.insertNodes(editor, image);
+    });
+  };
+
+  const handleSelectAssets = (images) => {
+    const formattedImages = images.map((image) => {
+      const expectedImage = pick(image, imageSchemaFields);
```

Comment:
we don't have types at the moment so this is hard to understand why you're doing it (no doubt you're doing it for a reason ofc lol) – would you be able to add a short comment explaining why please?